src/librustdoc/clean/types.rs

```diff
@@ -638,6 +607,50 @@ impl Attributes {
             .filter_map(clean_attr)
             .collect();
 
+        trait SingleExt {
+            type Item;
+            fn single(self) -> Option<Self::Item>;
+        }
+
+        impl<T: IntoIterator> SingleExt for T {
+            type Item = T::Item;
+            fn single(self) -> Option<Self::Item> {
+                let mut iter = self.into_iter();
+                let item = iter.next()?;
+                iter.next().is_none().then_some(())?;
```

Comment:
I think this is a little too cute :P
```suggestion
                if iter.next().is_some() {
                    return None;
                }
```

Reply:
I think I'll likely scrap the whole `SingleExt` extension trait as the diagnostics for 0 and >1 items should be different.